server/asset/eth/eth.go

```diff
@@ -189,7 +209,34 @@ func (eth *Backend) ValidateSecret(secret, contract []byte) bool {
 
 // Synced is true if the blockchain is ready for action.
 func (eth *Backend) Synced() (bool, error) {
-	return false, notImplementedErr
+	// node.SyncProgress will return nil both before syncing has begun and
+	// after it has finished. In order to discern when syncing has begun,
+	// wait for at least one change in block count, then defer to
+	// node.SyncProgress for the lifetime of the eth backend.
```

Comment:
> What is geth doing before sync?

On testnet at least, looking for peers for a long time. And I believe there is a window between that and sync starting, checking the node or whatever.

> There must be a better indicator.

Searching...

Reply:
I ask because perhaps we can key off of `peers` length as well.  If it's zero and syncprogress returns nil, assume not in sync.  If it's non-zero peer count and syncprogress is nil, assume in sync. 
There may be similar rpcs that indicate early startup stages as well.

Reply:
Could also ask in the go-ethereum repo

Reply:
Is there a way to ask questions besides making an issue? I asked in their discord, and someone suggested checking the block. We could use an external api for that...

But I think we can also use the txpool, or mempool. It seems there will be nothing in there before syncing is finished. So, if something is in there, we can fail fast, falling back to blocks for simnet or on the off chance eth has nothing in its mempool, which seems unlikely on main or testnets.

Reply:
I think I will just use peers.

Reply:
> I asked in their discord, and someone suggested checking the block.

What does this mean, "checking the block"?  Which block, and check for what?

> I think we can also use the txpool

That's something to think about.  Let's see how peers works out first.

Reply:
> What does this mean, "checking the block"? Which block, and check for what?

The only reply I got:
```
before syncing has begun = before it has started to pull any blocks at all?
if so, i guess you could try to check the block number as well or something, idk
```
idk if any of the devs are watching the discord, there may be a better place to ask

Reply:
Oh their response sounds like looking for initial sync rather than just startup at some unknown height.  Oh well.